### PR TITLE
kubelet: extract multiple ignore errors validate logic to isExpectedError

### PR DESCRIPTION
**Pull request by jonyhy96**
#### What type of PR is this?

/kind cleanup
/kind feature

#### What this PR does / why we need it:

Same error validate logic and comment has appeared multiple times 
```golang
	if err != nil &&
		!nestedpendingoperations.IsAlreadyExists(err) &&
		!exponentialbackoff.IsExponentialBackoff(err) {
		// Ignore nestedpendingoperations.IsAlreadyExists and exponentialbackoff.IsExponentialBackoff errors, they are expected.
		// Log all other errors.

```

Better extract them into a public function call

#### Special notes for your reviewer:

None

### CODE REVIEW

**Comment by jonyhy96** [comment]
maybe /sig-storage

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 1]
```diff
@@ -729,3 +715,8 @@ func getVolumesFromPodDir(podDir string) ([]podVolume, error) {
 	klog.V(4).InfoS("Get volumes from pod directory", "path", podDir, "volumes", volumes)
 	return volumes, nil
 }
+
+// ignore nestedpendingoperations.IsAlreadyExists and exponentialbackoff.IsExponentialBackoff errors, they are expected.
+func ignoreError(err error) bool {
```
Comment (endocrimes) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go:720]:
```suggestion
func isExpectedError(err error) bool {
```

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 2]
```diff
@@ -185,12 +185,10 @@ func (rc *reconciler) unmountVolumes() {
 			klog.V(5).InfoS(mountedVolume.GenerateMsgDetailed("Starting operationExecutor.UnmountVolume", ""))
 			err := rc.operationExecutor.UnmountVolume(
 				mountedVolume.MountedVolume, rc.actualStateOfWorld, rc.kubeletPodsDir)
-			if err != nil &&
-				!nestedpendingoperations.IsAlreadyExists(err) &&
-				!exponentialbackoff.IsExponentialBackoff(err) {
-				// Ignore nestedpendingoperations.IsAlreadyExists and exponentialbackoff.IsExponentialBackoff errors, they are expected.
-				// Log all other errors.
-				klog.ErrorS(err, mountedVolume.GenerateErrorDetailed(fmt.Sprintf("operationExecutor.UnmountVolume failed (controllerAttachDetachEnabled %v)", rc.controllerAttachDetachEnabled), err).Error())
+			if err != nil {
```
Comment (endocrimes) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go:188]: Rather than nesting this could you `if err != nil && !isExpectedError(err)`?

Comment (jonyhy96) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go:188]: Sure

File: pkg/kubelet/volumemanager/reconciler/reconciler.go (hunk 1 above)
Comment (jonyhy96) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go]: Thanks

**Review by endocrimes — APPROVED** [review]
/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**